### PR TITLE
[actions] Notification Builder: Add support for updating/hiding notifications

### PR DESCRIPTION
**Pull request by florian-h05**
Follow-up for #351.
Refs https://github.com/openhab/openhab-addons/issues/16934.

- Renames severity to tag.
- Adds support for specifying `referenceId`.
- Adds support for hiding notifications.

### CODE REVIEW

**Comment by florian-h05** [comment]
@digitaldan Can you please take a look soon, so we get this into 4.2.0?

**Comment by digitaldan** [comment]
Yes!!!! I will review in just a bit , very excited to use this on my own system .   Thanks for the quick turnaround!!!

File: src/actions/notification-builder.js [hunk 1]
```diff
@@ -160,27 +190,61 @@ class NotificationBuilder {
...
     while (this.#actionButtons.length < 3) {
       this.#actionButtons.push(null);
     }
+
     switch (this.#type) {
       case NotificationType.BROADCAST:
-        // parameters: message, icon, severity, title, onClickAction, mediaAttachmentUrl, actionButton1, actionButton2, actionButton3
-        _getNotificationAction()?.sendBroadcastNotification(this.#message, this.#icon, this.#severity, this.#title, this.#onClickAction, this.#mediaAttachmentUrl, ...this.#actionButtons);
-        break;
+        if (this.#referenceId === null) this.#referenceId = randomUUID();
```
Comment (digitaldan) [inline src/actions/notification-builder.js:206]: Any reason for not just letting the referenceId be null instead of setting it to  random value?

Comment (digitaldan) [reply src/actions/notification-builder.js:206]: also  now when i run it, i'm getting this error. I think this is b/c this is setting to a `java.util.UUID` object  and not an actual string. 

```
Failed to execute script: TypeError: invokeMember (sendBroadcastNotification) on org.openhab.io.openhabcloud.NotificationAction failed due to: no applicable overload found (overloads: [Method[public static void org.openhab.io.openhabcloud.NotificationAction.sendBroadcastNotification(java.lang.String)], Method[public static void org.openhab.io.openhabcloud.NotificationAction.sendBroadcastNotification(java.lang.String,java.lang.String,java.lang.String)], Method[public static void org.openhab.io.openhabcloud.NotificationAction.sendBroadcastNotification(java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String)]], arguments: [Test Message (String), f7:bell_fill (String), important (String), Important Notification (String), JavaObject[b6010bfd-b2f6-4e96-bc76-cb241287de7c (java.util.UUID)] (HostObject), DynamicObject<null>@3c2604c8 (Nullish), DynamicObject<null>@3c2604c8 (Nullish), DynamicObject<null>@3c2604c8 (Nullish), DynamicObject<null>@3c2604c8 (Nullish), DynamicObject<null>@3c2604c8 (Nullish)])
```

Comment (digitaldan) [reply src/actions/notification-builder.js:206]: Ah i see what you are doing, very cleaver! You are returning the ID for use later by a rule.  Nice 👍  i did not put together we could do that now.

Comment (digitaldan) [inline src/actions/notification-builder.js:206]:
```suggestion
        if (this.#referenceId === null) this.#referenceId = randomUUID().toString();
```

File: src/actions/notification-builder.js [hunk 2]
```diff
@@ -160,27 +190,61 @@ class NotificationBuilder {
...
+        return this.#referenceId;
       case NotificationType.LOG:
-        // parameters: message, icon, severity
-        _getNotificationAction()?.sendLogNotification(this.#message, this.#icon, this.#severity);
-        break;
+        // parameters: message, icon, tag
+        _getNotificationAction()?.sendLogNotification(this.#message, this.#icon, this.#tag);
+        return null;
       case NotificationType.STANDARD:
+        if (this.#referenceId === null) this.#referenceId = randomUUID();
```
Comment (digitaldan) [inline src/actions/notification-builder.js:215]:
```suggestion
        if (this.#referenceId === null) this.#referenceId = randomUUID().toString();
```

**Comment by florian-h05** [comment]
@digitaldan Thanks for your review.
Instead of adding `.toString()` where suggested, I fixed the randomUUID method which should return a string according to its JSDoc.

**Pull request merged**